### PR TITLE
Add support of user's primary Google Calendar events and theirs realtime changes

### PR DESCRIPTION
**Pull request by Vadimur**

### CODE REVIEW

File: src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            var updatedEvents = await Task.WhenAll(changedCalendarEvents
+                .Where(e => e.Status != "cancelled")
+                .Select(async e => await _eventConverter.ToCalendarEventEntity(e, calendarId)));
+
+            await _calendarEventsRepository.SaveRangeAsync(updatedEvents);
+
+            return true;
+        }
+        catch (Exception exception)
+        {
```
Comment (FairyFox5700) [inline src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs:166]: log?

File: src/Tracking/DisasterTrackerApp.Dal/Repositories/Implementation/GoogleUserRepository.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            return;
+        }
+        
+        user.AccessToken = newAccessToken;
+        user.ExpiresIn = DateTimeOffsetHelper.CalculateTokenExpirationTime(expiresIn);
+
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task UpdateTokensAsync(Guid userId, string newAccessToken, DateTimeOffset expiresAt, string newRefreshToken)
```
Comment (FairyFox5700) [inline src/Tracking/DisasterTrackerApp.Dal/Repositories/Implementation/GoogleUserRepository.cs:49]: single responsibility principle...
but ok

**Review by FairyFox5700 — COMMENTED** [review]
LG

File: src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            var eventsToDelete = changedCalendarEvents
+                .Select(e => e.Id)
+                .Intersect(existingCalendarEvents.Select(e => e.GoogleEventId))
+                .Select(id => existingCalendarEvents.First(e => e.GoogleEventId == id));
+
+            await _calendarEventsRepository.RemoveUserEventsAsync(eventsToDelete);
+
+            await RemoveOutdatedCalendarEvents(existingCalendarEvents);
+
+            var updatedEvents = await Task.WhenAll(changedCalendarEvents
```
Comment (FairyFox5700) [inline src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs:157]: image

File: src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs (hunk 1 above)
Comment (Vadimur) [reply src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs:166]: done

File: src/Tracking/DisasterTrackerApp.Dal/Repositories/Implementation/GoogleUserRepository.cs (hunk 2 above)
Comment (Vadimur) [reply src/Tracking/DisasterTrackerApp.Dal/Repositories/Implementation/GoogleUserRepository.cs:49]: User's tokens were part of user entity in initial database schema, so it is as it is. But totally agree, user info and his tokens should be separate entites. An improvement point for future versions.

File: src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs (hunk 3 above)
Comment (Vadimur) [reply src/Tracking/DisasterTrackerApp.BL/Implementation/UsersGoogleCalendarDataUpdatingService.cs:157]: Fair point. 
An improvement point for future versions.
Share link please

**Pull request merged**